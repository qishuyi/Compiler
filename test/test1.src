if 10 <= 8 then 8 / 2 + 6 else 2 * 6 - 10