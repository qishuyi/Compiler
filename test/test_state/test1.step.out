(let x = (ref (> 20 10)) in (x := (<= 20 10)) ; (if !x (- 2 1) (+ -1 1)))
(let x = Ptr(0) in (x := (<= 20 10)) ; (if !x (- 2 1) (+ -1 1)))
(Ptr(0) := (<= 20 10)) ; (if !Ptr(0) (- 2 1) (+ -1 1))
(Ptr(0) := false) ; (if !Ptr(0) (- 2 1) (+ -1 1))
() ; (if !Ptr(0) (- 2 1) (+ -1 1))
() ; (if false (- 2 1) (+ -1 1))
() ; 0
0
